src/jit/gentree.h

```diff
@@ -1909,15 +1916,55 @@ struct GenTree
     {
         return (gtFlags & GTF_REG_VAL) ? true : false;
     }
-    void SetInReg()
+    void SetInReg(bool value = true)
     {
-        gtFlags |= GTF_REG_VAL;
+        if (value == true)
+        {
+            gtFlags |= GTF_REG_VAL;
+        }
+        else
+        {
+            gtFlags &= ~GTF_REG_VAL;
+        }
     }
-
     regNumber GetReg() const
     {
         return InReg() ? gtRegNum : REG_NA;
     }
+
+#else // !LEGACY_BACKEND
+    // For the non-legacy backend, these are only used for dumping.
+    // The gtRegNum is only valid in LIR, but the dumping methods are not easily
+    // modified to check this.
+#ifdef DEBUG
+    bool InReg() const
+    {
+        return (GetRegTag() != GT_REGTAG_NONE) ? true : false;
+    }
+    regNumber GetReg() const
+    {
+        return (GetRegTag() != GT_REGTAG_NONE) ? gtRegNum : REG_NA;
+    }
+#endif
+
+    static bool IsContained(unsigned flags)
+    {
+        return ((flags & GTF_CONTAINED) != 0);
+    }
+
+    void SetAsContained()
+    {
+        gtFlags |= GTF_CONTAINED;
+    }
+
+    void ResetContained()
+    {
+        gtFlags &= ~GTF_CONTAINED;
+        gtLsraInfo.regOptional = false;
```

Comment:
Thanks for the link, it looks like I missed or completely forgot about that document